# Mutual Funds Analysis

## Project Definition (July 15, 2026)

### Business Problem

Financial advisors and retail investors often need to compare thousands of mutual funds with different performance histories, risk levels, fees, and investment requirements. The volume and complexity of available information makes it difficult to evaluate funds efficiently and consistently. A structured data analysis can uncover relationships among key fund characteristics and generate insights that support more informed investment decisions.

### Project Objective

Analyze a comprehensive mutual funds dataset using Python and Pandas to explore fund characteristics, compare performance, and generate data-driven insights that support more informed investment decisions.

### Initial Business Questions

1. How do mutual fund performance metrics vary across different funds and investment categories?
2. What relationships exist between fund characteristics and historical performance?
3. How do risk-related measures compare with historical fund performance?
4. How do investment requirements differ across mutual funds?
5. What insights can help financial advisors and retail investors compare mutual funds more effectively?

### Tools

- Python
- Pandas
- Jupyter Notebook

### Dataset

This project uses a publicly available mutual funds dataset containing information on **23,783 mutual funds** and **298 variables**, including fund characteristics, historical performance, risk measures, investment requirements, and other financial metrics.

In [1]:
import pandas as pd

# Load the dataset
df = pd.read_csv("../data/MutualFunds.csv")

In [2]:
df.head()

,fund_symbol,quote_type,region,fund_short_name,fund_long_name,currency,initial_investment,subsequent_investment,fund_category,fund_family,...,peer_environment_avg,peer_environment_max,social_score,peer_social_min,peer_social_avg,peer_social_max,governance_score,peer_governance_min,peer_governance_avg,peer_governance_max
0,AAAAX,MutualFund,US,DWS RREEF Real Assets Fund - Cl,DWS RREEF Real Assets Fund - Class A,USD,1000.0,50.0,World Allocation,DWS,...,5.05,10.58,7.43,5.98,9.07,11.30,5.43,4.26,7.14,8.11
1,AAAEX,MutualFund,US,AllianzGI Health Sciences Fund,Virtus AllianzGI Health Sciences Fund Class P,USD,1000000.0,NaN,Health,Virtus,...,1.43,3.27,12.96,9.52,12.87,15.08,8.40,4.96,7.68,10.30
2,AAAFX,MutualFund,US,NaN,American Century One Choice Blend+ 2015 Portfo...,USD,2500.0,50.0,Target-Date 2015,American Century Investments,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,AAAGX,MutualFund,US,Thrivent Large Cap Growth Fund,Thrivent Large Cap Growth Fund Class A,USD,2000.0,50.0,Large Growth,Thrivent Funds,...,2.70,5.81,10.13,7.25,10.14,11.97,8.03,5.30,7.54,8.90
4,AAAHX,MutualFund,US,NaN,American Century One Choice Blend+ 2015 Portfo...,USD,5000000.0,NaN,Target-Date 2015,American Century Investments,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [3]:
df.shape

(23783, 298)

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23783 entries, 0 to 23782
Columns: 298 entries, fund_symbol to peer_governance_max
dtypes: float64(277), object(21)
memory usage: 54.1+ MB


In [5]:
df.describe()

,initial_investment,subsequent_investment,total_net_assets,year_to_date_return,day50_moving_average,day200_moving_average,week52_high_low_change,week52_high_low_change_perc,week52_high,week52_high_change,...,peer_environment_avg,peer_environment_max,social_score,peer_social_min,peer_social_avg,peer_social_max,governance_score,peer_governance_min,peer_governance_avg,peer_governance_max
count,1.648500e+04,8943.000000,2.374900e+04,23382.000000,23782.000000,23782.000000,23496.000000,23496.000000,23496.000000,23421.000000,...,14839.000000,14839.000000,15388.000000,14839.000000,14839.000000,14839.000000,15388.000000,14839.000000,14839.000000,14839.000000
mean,5.179181e+06,1015.925752,4.924192e+09,0.093485,23.638731,23.084989,4.643490,0.146057,24.995830,-1.957576,...,4.357856,7.675547,8.043626,6.906983,9.346850,11.308239,6.210819,5.387539,7.219275,8.917380
std,9.253584e+07,11023.969484,2.109765e+10,0.082276,33.126315,31.552946,9.312633,0.092420,35.208749,4.069716,...,1.498273,2.838513,3.674170,1.167251,1.204409,1.732701,2.800713,0.748176,0.820705,1.364138
min,1.000000e+01,1.000000,1.000000e+01,-0.522800,1.279000,1.363000,0.000000,0.000000,1.700000,-101.420040,...,0.750000,0.830000,0.000000,2.300000,4.410000,5.860000,0.000000,2.610000,3.040000,3.310000
25%,1.000000e+03,50.000000,1.445834e+08,0.026100,10.997250,10.968000,0.750000,0.061890,11.350000,-1.960000,...,3.810000,5.810000,8.240000,6.530000,9.060000,10.710000,6.230000,5.050000,6.930000,7.950000
50%,2.500000e+03,50.000000,6.443283e+08,0.092000,14.959500,14.709000,2.420000,0.149255,15.660000,-0.800000,...,4.080000,7.420000,9.480000,7.040000,9.450000,11.520000,7.350000,5.300000,7.420000,8.950000
75%,5.000000e+05,100.000000,2.453409e+09,0.149700,25.184500,24.693250,5.390000,0.211635,26.680000,-0.340000,...,5.000000,9.890000,10.190000,7.420000,10.120000,12.040000,7.740000,5.830000,7.540000,9.650000
max,5.000000e+09,500000.000000,7.534100e+11,0.578900,2018.338000,1920.319000,455.750000,0.882960,2092.820000,0.004900,...,14.350000,21.290000,15.080000,14.780000,14.790000,17.250000,14.020000,11.650000,11.650000,15.300000


In [6]:
df["initial_investment"].describe()

count    1.648500e+04
mean     5.179181e+06
std      9.253584e+07
min      1.000000e+01
25%      1.000000e+03
50%      2.500000e+03
75%      5.000000e+05
max      5.000000e+09
Name: initial_investment, dtype: float64

## Initial Dataset Exploration (July 15, 2026)

### Dataset Overview
- The dataset contains **23,783 mutual fund records** and **298 variables**.
- Most variables are numerical (**277 float columns**), while **21 columns** contain categorical or text data.

### Data Quality
- The `initial_investment` column contains missing values and will require further investigation.
- A more comprehensive missing value analysis will be performed across the full dataset during the data quality assessment phase.

### Initial Findings
- The `initial_investment` distribution appears to be highly right-skewed.
- The mean initial investment (approximately **$5.18 million**) is substantially higher than the median (**$2,500**), suggesting that a relatively small number of funds with very large investment requirements are influencing the average.

### Questions for Further Analysis
- What exactly does the `initial_investment` column represent?
- Which funds have the highest initial investment requirements?
- Are the extreme values legitimate observations or potential outliers?
- How many missing values exist in each column?

In [7]:
df.isnull().sum()

fund_symbol               0
quote_type                0
region                    0
fund_short_name         824
fund_long_name            5
                       ... 
peer_social_max        8944
governance_score       8395
peer_governance_min    8944
peer_governance_avg    8944
peer_governance_max    8944
Length: 298, dtype: int64

In [8]:
missing_values = df.isnull().sum().sort_values(ascending=False)
missing_values.head(15)

last_cap_gain                    23776
fund_return_2021_q3              23721
fund_max_deferred_sales_load     20780
fund_return_2021_q2              20166
fund_max_front_end_sales_load    20090
fund_return_2000_q4              19827
fund_return_2000                 19827
fund_return_2000_q3              19734
fund_return_2000_q2              19599
fund_return_2000_q1              19462
fund_return_2001                 19278
fund_return_2001_q4              19278
fund_return_2001_q3              19051
fund_return_2001_q2              18926
fund_return_2001_q1              18797
dtype: int64

## Missing Values Assessment (July 24, 2026)

### Observations

- Missing values are present across several variables, but the extent of missingness varies considerably throughout the dataset.
- Variables with the highest percentages of missing values are primarily related to historical fund returns, sales load information, and optional fund characteristics.
- The `last_cap_gain` variable is missing for nearly all observations (approximately 99.97%) and is unlikely to provide meaningful analytical value.
- Historical return variables from the early 2000s contain substantial missing data, likely because many mutual funds in the dataset were established after that period and therefore do not have performance records for those years.
- The key variables selected for this project (fund category, expense ratio, returns, Morningstar ratings, and total net assets) have relatively low levels of missing data and are suitable for exploratory analysis and visualization.
- Based on the initial assessment, no variables will be removed at this stage. Columns with extensive missing values will simply be excluded from analyses where they are not relevant.

In [9]:
# Check for duplicate rows
duplicate_rows = df.duplicated().sum()

print(f"Duplicate rows: {duplicate_rows}")

Duplicate rows: 0


In [10]:
# Check duplicate fund symbols
duplicate_funds = df["fund_symbol"].duplicated().sum()

print(f"Duplicate fund symbols: {duplicate_funds}")

Duplicate fund symbols: 0


In [11]:
missing_values = df.isnull().sum()

missing_summary = (
    pd.DataFrame({
        "Missing Values": missing_values,
        "Percentage": (missing_values / len(df) * 100)
    })
    .sort_values("Percentage", ascending=False)
)

missing_summary.head(20)

,Missing Values,Percentage
last_cap_gain,23776,99.970567
fund_return_2021_q3,23721,99.739310
fund_max_deferred_sales_load,20780,87.373334
fund_return_2021_q2,20166,84.791658
fund_max_front_end_sales_load,20090,84.472102
fund_return_2000_q4,19827,83.366270
fund_return_2000,19827,83.366270
fund_return_2000_q3,19734,82.975234
fund_return_2000_q2,19599,82.407602
fund_return_2000_q1,19462,81.831560


In [12]:
df.columns.tolist()

['fund_symbol',
 'quote_type',
 'region',
 'fund_short_name',
 'fund_long_name',
 'currency',
 'initial_investment',
 'subsequent_investment',
 'fund_category',
 'fund_family',
 'exchange_code',
 'exchange_name',
 'exchange_timezone',
 'management_name',
 'management_bio',
 'management_start_date',
 'total_net_assets',
 'year_to_date_return',
 'day50_moving_average',
 'day200_moving_average',
 'week52_high_low_change',
 'week52_high_low_change_perc',
 'week52_high',
 'week52_high_change',
 'week52_high_change_perc',
 'week52_low',
 'week52_low_change',
 'week52_low_change_perc',
 'investment_strategy',
 'fund_yield',
 'morningstar_overall_rating',
 'morningstar_risk_rating',
 'inception_date',
 'last_dividend',
 'last_cap_gain',
 'annual_holdings_turnover',
 'investment_type',
 'size_type',
 'fund_annual_report_net_expense_ratio',
 'category_annual_report_net_expense_ratio',
 'fund_prospectus_net_expense_ratio',
 'fund_prospectus_gross_expense_ratio',
 'fund_max_12b1_fee',
 'fund_max_f

In [13]:
key_columns = [
    "fund_symbol",
    "fund_category",
    "fund_prospectus_net_expense_ratio",
    "fund_return_1year",
    "fund_return_3years",
    "fund_return_5years",
    "morningstar_overall_rating",
    "morningstar_risk_rating",
    "total_net_assets",
    "initial_investment"
]

df[key_columns].isnull().sum()

fund_symbol                             0
fund_category                         663
fund_prospectus_net_expense_ratio     216
fund_return_1year                     896
fund_return_3years                   2697
fund_return_5years                   6760
morningstar_overall_rating           1807
morningstar_risk_rating              1807
total_net_assets                       34
initial_investment                   7298
dtype: int64

## Data Quality Assessment (July 24, 2026)

### Duplicate Rows
No duplicate rows were identified in the dataset, indicating that each record is unique.

### Duplicate Funds
No duplicate fund symbols were found. Each fund is represented by a single unique record.

### Missing Values
Several variables contain high percentages of missing values. Most of these columns correspond to historical performance data, sales load information, or optional fund characteristics. The missing values appear to reflect the nature of the financial data rather than data quality issues.

### Key Variables
The primary variables selected for analysis have relatively low levels of missing data. Core fields such as fund symbol, fund category, expense ratio, one-year return, Morningstar ratings, and total net assets are sufficiently complete for analysis. Higher missing rates in the five-year return and initial investment columns are expected because newer funds may not have a five-year performance history, and some funds do not report minimum investment requirements.

### Selected Variables for Analysis
The analysis will primarily focus on the following variables:

- **Fund Identification:** `fund_symbol`, `fund_short_name`, `fund_category`
- **Fund Characteristics:** `investment_type`, `size_type`, `total_net_assets`
- **Cost:** `fund_prospectus_net_expense_ratio`
- **Performance:** `fund_return_1year`, `fund_return_3years`, `fund_return_5years`, `fund_return_10years`
- **Risk:** `morningstar_overall_rating`, `morningstar_risk_rating`
- **Portfolio Allocation:** `asset_stocks`, `asset_bonds`, `asset_cash`

### Conclusion
The dataset is well-structured and suitable for exploratory data analysis. No duplicate records or duplicate fund symbols were identified, and the key variables selected for this project are sufficiently complete to support meaningful analysis and visualization. While some variables contain substantial missing values, these are primarily associated with historical or optional fund information and are not expected to significantly impact the objectives of this project.

In [14]:
important_columns = [
    "total_net_assets",
    "initial_investment",
    "fund_prospectus_net_expense_ratio",
    "fund_return_1year",
    "fund_return_3years",
    "fund_return_5years"
]

df[important_columns].describe()

,total_net_assets,initial_investment,fund_prospectus_net_expense_ratio,fund_return_1year,fund_return_3years,fund_return_5years
count,2.374900e+04,1.648500e+04,23567.000000,22887.000000,21086.000000,17023.000000
mean,4.924192e+09,5.179181e+06,0.010473,0.239720,0.101469,0.081893
std,2.109765e+10,9.253584e+07,0.005793,0.177877,0.071840,0.056171
min,1.000000e+01,1.000000e+01,0.000100,-0.637900,-0.512700,-0.429600
25%,1.445834e+08,1.000000e+03,0.006600,0.080500,0.055800,0.042600
50%,6.443283e+08,2.500000e+03,0.009500,0.250500,0.091100,0.078300
75%,2.453409e+09,5.000000e+05,0.013400,0.362500,0.135300,0.112500
max,7.534100e+11,5.000000e+09,0.118000,1.350300,0.648240,0.448580


In [15]:
df["fund_category"].value_counts().head(15)

fund_category
Large Blend                      1235
Large Growth                     1172
Large Value                      1080
Diversified Emerging Mkts         741
Foreign Large Blend               680
Allocation--50% to 70% Equity     658
High Yield Bond                   629
Small Blend                       579
Small Growth                      578
Short-Term Bond                   570
Intermediate Core-Plus Bond       566
Mid-Cap Growth                    562
Allocation--30% to 50% Equity     491
World Allocation                  448
Foreign Large Growth              431
Name: count, dtype: int64

In [16]:
df["investment_type"].value_counts()

investment_type
Blend     10902
Value      6847
Growth     4925
Name: count, dtype: int64

In [17]:
df["size_type"].value_counts()

size_type
Large     13199
Medium     5582
Small      3893
Name: count, dtype: int64

In [18]:
df["morningstar_overall_rating"].value_counts().sort_index()

morningstar_overall_rating
1.0    1620
2.0    4643
3.0    7911
4.0    5450
5.0    2352
Name: count, dtype: int64

In [19]:
for col in [
    "total_net_assets",
    "initial_investment",
    "fund_prospectus_net_expense_ratio"
]:
    print("=" * 50)
    print(col)
    print(df[col].describe())

total_net_assets
count    2.374900e+04
mean     4.924192e+09
std      2.109765e+10
min      1.000000e+01
25%      1.445834e+08
50%      6.443283e+08
75%      2.453409e+09
max      7.534100e+11
Name: total_net_assets, dtype: float64
initial_investment
count    1.648500e+04
mean     5.179181e+06
std      9.253584e+07
min      1.000000e+01
25%      1.000000e+03
50%      2.500000e+03
75%      5.000000e+05
max      5.000000e+09
Name: initial_investment, dtype: float64
fund_prospectus_net_expense_ratio
count    23567.000000
mean         0.010473
std          0.005793
min          0.000100
25%          0.006600
50%          0.009500
75%          0.013400
max          0.118000
Name: fund_prospectus_net_expense_ratio, dtype: float64


## Exploratory Data Analysis (July 24, 2026)

### Distribution of Key Variables

- The financial variables `total_net_assets` and `initial_investment` exhibit strongly right-skewed distributions, indicating that a relatively small number of funds have exceptionally large asset values or minimum investment requirements.
- Expense ratios are much more concentrated, with most funds reporting prospectus net expense ratios below 1.5%.
- One-year returns show greater variability than three-year and five-year returns, while longer-term returns appear more stable.

### Categorical Variables

- The dataset contains a wide variety of mutual fund categories, with **Large Blend**, **Large Growth**, and **Large Value** being the most common.
- **Blend** funds represent the largest investment type, followed by **Value** and **Growth** funds.
- Most funds are classified as **Large-cap**, followed by **Medium-cap** and **Small-cap** funds.
- Morningstar overall ratings are concentrated around **3 and 4 stars**, with fewer funds receiving the highest or lowest ratings.

### Outliers

- Extreme values are present in both `total_net_assets` and `initial_investment`.
- These observations appear to represent legitimate characteristics of very large mutual funds rather than data entry errors, so they will be retained for analysis.

### Variables Selected for Visualization

The following variables were selected for future visualizations because they are relevant to the project's objectives and contain sufficient data:

- Fund category (`fund_category`)
- Investment type (`investment_type`)
- Fund size (`size_type`)
- Total net assets (`total_net_assets`)
- Prospectus net expense ratio (`fund_prospectus_net_expense_ratio`)
- One-year, three-year, and five-year returns
- Morningstar overall rating (`morningstar_overall_rating`)
- Asset allocation (`asset_stocks`, `asset_bonds`, `asset_cash`)

### Conclusion

The exploratory analysis identified several meaningful patterns in the dataset, including right-skewed financial variables, diverse fund categories, and a broad range of performance and risk characteristics. The selected variables provide a strong foundation for the visualization and comparative analyses that follow.

In [20]:
columns_to_keep = [
    "fund_symbol",
    "fund_category",
    "investment_type",
    "size_type",
    "total_net_assets",
    "initial_investment",
    "fund_prospectus_net_expense_ratio",
    "fund_return_1year",
    "fund_return_3years",
    "fund_return_5years",
    "morningstar_overall_rating",
    "morningstar_risk_rating",
    "asset_stocks",
    "asset_bonds",
    "asset_cash"
]

print(columns_to_keep)

['fund_symbol', 'fund_category', 'investment_type', 'size_type', 'total_net_assets', 'initial_investment', 'fund_prospectus_net_expense_ratio', 'fund_return_1year', 'fund_return_3years', 'fund_return_5years', 'morningstar_overall_rating', 'morningstar_risk_rating', 'asset_stocks', 'asset_bonds', 'asset_cash']


## Data Cleaning Plan
**Date:** July 27, 2026

### 1. Key Variables Selected for Initial Analysis

The following variables were selected because they are directly related to the project's business questions, contain sufficient data for meaningful analysis, and represent the most important aspects of mutual fund evaluation.

| Variable | Reason for Selection |
|----------|----------------------|
| `fund_symbol` | Unique identifier for each mutual fund. |
| `fund_category` | Compare different categories of mutual funds. |
| `investment_type` | Compare Blend, Growth, and Value funds. |
| `size_type` | Compare funds by market capitalization. |
| `total_net_assets` | Measure fund size and popularity. |
| `initial_investment` | Analyze investor accessibility and investment requirements. |
| `fund_prospectus_net_expense_ratio` | Evaluate fund costs. |
| `fund_return_1year` | Measure recent performance. |
| `fund_return_3years` | Measure medium-term performance. |
| `fund_return_5years` | Measure longer-term performance. |
| `morningstar_overall_rating` | Compare independent fund ratings. |
| `morningstar_risk_rating` | Compare relative investment risk. |
| `asset_stocks` | Measure percentage invested in stocks. |
| `asset_bonds` | Measure percentage invested in bonds. |
| `asset_cash` | Measure percentage invested in cash. |

---

### 2. Column Inventory (Version 1)

The dataset contains 298 variables. Rather than immediately deleting columns, the variables were grouped according to their analytical purpose to determine which should be prioritized, retained for future analysis, or considered for removal after further review.

| Column Group | Example Columns | Keep for Analysis? | Remove Now? | Reason |
|--------------------------|------------------------------------------------------------------------------------------|------------------|-------------|---------------------------------------------------------------------------------|
| Identification | `fund_symbol`, `fund_short_name`, `fund_long_name` | ✅ Yes | ❌ No | Needed to identify and reference individual mutual funds. |
| Fund Classification | `fund_category`, `investment_type`, `size_type` | ✅ Yes | ❌ No | Essential for comparing groups of funds. |
| Investment Requirements | `initial_investment`, `subsequent_investment` | ✅ Yes | ❌ No | Important for understanding investor accessibility. |
| Assets & Fund Size | `total_net_assets` | ✅ Yes | ❌ No | Key measure of fund size and popularity. |
| Fees & Expenses | `fund_prospectus_net_expense_ratio`, `fund_prospectus_gross_expense_ratio` | ✅ Yes | ❌ No | Directly relevant to evaluating fund costs. |
| Performance (Summary) | `fund_return_1year`, `fund_return_3years`, `fund_return_5years`, `year_to_date_return` | ✅ Yes | ❌ No | Core performance metrics that answer the project's business questions. |
| Historical Annual Returns | `fund_return_2000`–`fund_return_2020` | 🟡 Maybe Later | ❌ No | Valuable for deeper historical analysis but not required in the initial phase. |
| Quarterly Returns | `fund_return_2000_q1`–`fund_return_2021_q3` | 🟡 Maybe Later | ❌ No | Very detailed information that may be explored later if needed. |
| Risk Metrics | `fund_alpha_3years`, `fund_beta_3years`, `fund_sharpe_ratio_3years`, `fund_stdev_3years` | 🟡 Maybe Later | ❌ No | Important financial metrics that may support advanced analysis. |
| Category Rankings | `fund_return_category_rank_*` | 🟡 Maybe Later | ❌ No | Useful for benchmarking but not essential for the initial analysis. |
| ESG Metrics | `sustainability_score`, `environment_score`, `social_score`, `governance_score` | 🟡 Maybe Later | ❌ No | Interesting information but outside the primary scope of this project. |
| Portfolio Composition | `asset_stocks`, `asset_bonds`, `asset_cash`, sector allocation columns | 🟡 Maybe Later | ❌ No | Can support deeper investment analysis in future phases. |
| Holdings | `top10_holdings`, `top10_holdings_total_assets` | 🟡 Maybe Later | ❌ No | Useful for understanding portfolio composition but not a priority. |
| Management Information | `management_name`, `management_bio`, `management_start_date` | ❌ No | ❌ No | Not needed for the current analysis but may have value for future research. |
| Sales Load Information | `fund_max_front_end_sales_load`, `fund_max_deferred_sales_load` | ❌ No | 🟡 Review | High percentage of missing values; assess usefulness before removal. |
| Rarely Populated Variables | `last_cap_gain` and similar columns with >95% missing values | ❌ No | ✅ Yes | Extremely high missingness makes them unlikely to contribute meaningful insights. |

---

### 3. Missing Value Strategy (Version 1)

| Situation | Planned Action | Reason |
|-----------|----------------|--------|
| Missing values in key analysis variables (e.g., `initial_investment`, `fund_return_1year`) | Keep the columns and decide how to handle missing values based on the specific analysis. Some rows may be excluded only when necessary. | These variables are essential to answering the project's business questions. |
| Columns with extremely high missingness (approximately >95%) | Consider removing the entire column after confirming it has little analytical value. | These columns contain too little information to support meaningful analysis. |
| Historical return variables with many missing values | Keep the columns for now. Missing values are likely due to newer funds lacking historical performance. | The missingness reflects the nature of the data rather than a data quality problem. |
| Optional fund characteristics (e.g., sales load information, ESG metrics) | Evaluate each group individually before deciding whether to keep or remove it. | These variables may be useful for future analyses even if they are not part of the initial project scope. |
| Missing categorical values | Investigate whether the missing values represent unavailable information or a meaningful category before taking action. | Avoid making assumptions about missing categorical data. |
| Missing numerical values | Do not automatically replace missing values. Choose an appropriate treatment based on the variable and the analysis being performed. | Different numerical variables require different handling strategies. |

---

### 4. Planned Cleaning Steps

1. Preserve all original data in the raw dataset.
2. Create a cleaned dataset instead of modifying the original data.
3. Remove only columns that have both extremely high missingness and little analytical value.
4. Handle missing values individually according to each analysis.
5. Verify data types before performing any transformations.
6. Document every cleaning decision to ensure reproducibility.